octoprint_mrbeam/service/profile/profile.py

```diff
@@ -0,0 +1,482 @@
+        if identifier is not None and not identifier in all_identifiers:
+            return
+
+        settings().set(["profiles", self._id, "default"], identifier, force=True)
+        settings().save()
+
+    def get_current_or_default(self):
+        if self._current is not None:
+            return self._current
+        else:
+            return self.get_default()
+
+    def get_current(self):
+        return self._current
+
+    def exists(self, identifier):
+        if identifier is None:
+            return False
+        else:
+            path = self._get_profile_path(identifier)
+            return os.path.exists(path) and os.path.isfile(path)
+
+    def _load_all(self):
+        all_identifiers = self._load_all_identifiers()
+        results = dict()
+        for identifier, path in all_identifiers.items():
+            try:
+                profile = self._load_from_path(path)
+            except InvalidProfileError:
+                self._logger.warn("Profile {} is invalid, skipping".format(identifier))
+                continue
+
+            if profile is None:
+                continue
+
+            results[identifier] = dict_merge(self._default, profile)
+        return results
+
+    def _load_all_identifiers(self):
+        results = dict()
+        for entry in scandir(self._folder):
+            if is_hidden_path(entry.name) or not entry.name.endswith(".profile"):
+                continue
+
+            if not entry.is_file():
+                continue
+
+            identifier = entry.name[:-len(".profile")]
+            results[identifier] = entry.path
+        return results
+
+    def _load_from_path(self, path):
+        if not os.path.exists(path) or not os.path.isfile(path):
+            return None
+
+        import yaml
+        with open(path) as f:
+            profile = yaml.safe_load(f)
+
+        if profile is None or not isinstance(profile, dict):
+            raise InvalidProfileError("Profile is None or not a dictionary")
+
+        if self._migrate_profile(profile):
+            try:
+                self._save_to_path(path, profile, allow_overwrite=True)
+            except:
+                self._logger.exception(
+                    "Tried to save profile to {path} after migrating it while loading, ran into exception".format(
+                        path=path))
+
+        profile = self._ensure_valid_profile(profile)
+
+        if not profile:
+            self._logger.warn("Invalid profile: %s" % path)
+            raise InvalidProfileError()
+        return profile
+
+    def _save_to_path(self, path, profile, allow_overwrite=False):
+        validated_profile = self._ensure_valid_profile(profile)
+        if not validated_profile:
+            raise InvalidProfileError()
+
+        if os.path.exists(path) and not allow_overwrite:
+            raise SaveError("Profile %s already exists and not allowed to overwrite" % profile["id"])
+
+        from octoprint.util import atomic_write
```

Comment:
Same with these imports

Reply:
This is not my implementation but OctoPrint's. We are using their class as a parent class with minor changes to fir our purpose